src/node.rs

```diff
@@ -46,3 +40,10 @@ pub struct NodeConfig {
 // impl WeakSignature for Node {}
 
 // impl StrongSignature for Node {}
+
+// ------ METRIC SPACE (distance) ------
+pub trait MetricSpace: Sized {
```

Comment:
I want to be able to identify the distance metric between identifiers separately from the `id` type itself. I want to provide default implementations 
* bitwise xor from `&[u8]` (for kademlia)
* chord's clockwise circle metric

Reply:
nah, I'm doing this a bit differently now; grouped `distance`, `is_zero` and some other method into trait `KadMetric`